src/odemis/gui/comp/viewport.py

```diff
@@ -781,17 +783,108 @@ class AngularResolvedViewport(ViewPort):
 
     # Default class
     canvas_class = miccanvas.AngularResolvedCanvas
-    bottom_legend_class = None
+    bottom_legend_class = RadioLegend
+
+    def __init__(self, *args, **kwargs):
+        super(AngularResolvedViewport, self).__init__(*args, **kwargs)
+        # TODO why not sufficient to place here?
+        # bottom_legend_class = RadioLegend
 
     def setView(self, view, tab_data):
+        """
+        Set the microscope view that this viewport is displaying/representing
+        *Important*: Should be called only once, at initialisation.
+
+        :param view:(model.View)
+        :param tab_data: (model.MicroscopyGUIData)
+        """
         assert(self._view is None)
 
         self._view = view
         self._tab_data_model = tab_data
+        self.stream = None
 
         # canvas handles also directly some of the view properties
         self.canvas.setView(view, tab_data)
 
+        view.stream_tree.flat.subscribe(self.connect_stream)
+
+        self.bottom_legend.set_pol_callback(self.on_pol_change)
+
+    def connect_stream(self, _):
```

Comment:
You actually could received the streams (`ss`) as the argument of the method, now that it's connected to `.flat`. This avoids calling `self._view.getStreams()` just after.